theano/gpuarray/opt.py

```diff
@@ -2069,7 +2085,7 @@ def provide_inputs(self, node, inputs):
                                          node.op.subsample,
                                          node.op.filter_dilation)
 
-            result[kshape] = theano.tensor.as_tensor_variable(node.op.kshp[2:])
+            result[kshape] = theano.tensor.as_tensor_variable(node.op.kshp[-2:])
```

Comment:
I didn't really follow the metaoptimizer work, so I don't know how difficult it would be to test this. But if the metaopimizer can select those optimizers, then I think it would be good to have a test. (Although probably not as part of this PR.)